pkg/reconciler/events/storage/storage.go

```diff
@@ -345,52 +244,18 @@ func (r *Reconciler) updateStatus(ctx context.Context, original *v1alpha1.CloudS
 	})
 }
 
-// updateFinalizers is a generic method for future compatibility with a
-// reconciler SDK.
-func (r *Reconciler) updateFinalizers(ctx context.Context, desired *v1alpha1.CloudStorageSource) (*v1alpha1.CloudStorageSource, bool, error) {
-	storage, err := r.storageLister.CloudStorageSources(desired.Namespace).Get(desired.Name)
-	if err != nil {
-		return nil, false, err
-	}
-
-	// Don't modify the informers copy.
-	existing := storage.DeepCopy()
-
-	var finalizers []string
-
-	// If there's nothing to update, just return.
-	existingFinalizers := sets.NewString(existing.Finalizers...)
-	desiredFinalizers := sets.NewString(desired.Finalizers...)
-
-	if desiredFinalizers.Has(finalizerName) {
-		if existingFinalizers.Has(finalizerName) {
-			// Nothing to do.
-			return desired, false, nil
-		}
-		// Add the finalizer.
-		finalizers = append(existing.Finalizers, finalizerName)
-	} else {
-		if !existingFinalizers.Has(finalizerName) {
-			// Nothing to do.
-			return desired, false, nil
-		}
-		// Remove the finalizer.
-		existingFinalizers.Delete(finalizerName)
-		finalizers = existingFinalizers.List()
-	}
-
-	mergePatch := map[string]interface{}{
-		"metadata": map[string]interface{}{
-			"finalizers":      finalizers,
-			"resourceVersion": existing.ResourceVersion,
-		},
+func (r *Reconciler) FinalizeKind(ctx context.Context, storage *v1alpha1.CloudStorageSource) reconciler.Event {
+	logging.FromContext(ctx).Desugar().Debug("Deleting CloudStorageSource notification")
+	if err := r.deleteNotification(ctx, storage); err != nil {
+		storage.Status.MarkNotificationNotReady("NotificationDeleteFailed", "Failed to delete CloudStorageSource notification: %s", err.Error())
```

Comment:
same as below... I think this MarkNotificationNotReady is not needed

Reply:
done